boot/seal.go

```diff
@@ -109,6 +115,26 @@ func sealKeyToModeenv(key secboot.EncryptionKey, model *asserts.Model, modeenv *
 	return nil
 }
 
+func stampSealedKeys(rootdir string) error {
+	stamp := filepath.Join(dirs.SnapFDEDirUnder(rootdir), "sealed-keys")
+	if err := os.MkdirAll(filepath.Dir(stamp), 0755); err != nil {
+		return fmt.Errorf("cannot create device fde state directory: %v", err)
+	}
+
+	if err := osutil.AtomicWriteFile(stamp, nil, 0644, 0); err != nil {
+		return fmt.Errorf("cannot create fde sealed keys stamp file: %v", err)
+	}
+	return nil
+}
+
+// hasSealedKeys return whether any keys were sealed at all
+func hasSealedKeys(rootdir string) bool {
+	// TODO:UC20: consider more than the marker for cases where we reseal
```

Comment:
What cases do we reseal outside of run mode?

Reply:
resealing from recovery (but it's a complex case)